src/lib/models/event-history.ts

```diff
@@ -1,24 +1,31 @@
+import { convertPayloadToJson } from '$lib/utilities/decode-payload';
 import type { GetWorkflowExecutionHistoryResponse, HistoryEvent } from '$types';
 
-export function getEventAttributes(historyEvent: HistoryEvent): EventAttribute {
+export async function getEventAttributes(
+  historyEvent: HistoryEvent,
+): Promise<EventAttribute> {
   const attributeKey = Object.keys(historyEvent).find((key) =>
     key.includes('Attributes'),
   );
+  const attributes = await convertPayloadToJson(historyEvent[attributeKey]);
+
   return {
     type: attributeKey as EventAttributeKeys,
-    ...historyEvent[attributeKey],
+    ...attributes,
   };
 }
 
-export const toEventHistory = (
+export const toEventHistory = async (
   response: GetWorkflowExecutionHistoryResponse,
-): HistoryEventWithId[] => {
-  return response.history.events.map(
-    (event): HistoryEventWithId => ({
-      ...event,
-      id: String(event.eventId),
-      eventType: event.eventType as unknown as EventType,
-      attributes: getEventAttributes(event),
-    }),
+): Promise<HistoryEventWithId[]> => {
+  return await Promise.all(
+    response.history.events.map(
+      async (event): Promise<HistoryEventWithId> => ({
+        ...event,
+        id: String(event.eventId),
+        eventType: event.eventType as unknown as EventType,
```

Comment:
This is not your problem, but maybe we just need to maintain our own types.

Reply:
Honestly the types were a huge issue for getting this right and it was easier to any several things because they were wrong.